### PR TITLE
Consolidate image package and make SIF image code more generic

### PR DESCRIPTION
**Pull request by cclerget**
**Description of the Pull Request (PR):**

This PR consolidate image package and makes SIF image code more generic to handle use case like plugin embedded in SIF image in #3123

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

File: pkg/image/image.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ const (
 
 const (
 	// RootFs partition name
-	RootFs       = "rootfs"
+	RootFs       = "!__rootfs__!"
```
Comment (mem) [inline pkg/image/image.go:32]: Everything looks good.

The one thing I'm not sure about is this change.

Does this have an impact on the images that get created? In particular, what happens with images created before this change? I see there's a new check using HasRootFs... are older images rejected?

Related to this: does the change on line 21 (SQUASHFS) have an impact on images?

File: pkg/image/image.go [hunk 2]
```diff
@@ -134,6 +134,16 @@ func (i *Image) AuthorizedGroup(groups []string) (bool, error) {
 	return authorized, nil
 }
 
+// HasRootfs returns if image contains a root filesystem partition
+func (i *Image) HasRootfs() bool {
```
Comment (mem) [inline pkg/image/image.go]: minor thing: maybe HasRootFs?

File: pkg/image/image.go (hunk 1 above)
Comment (cclerget) [reply pkg/image/image.go:32]: No impact, those informations are used by runtime only to somewhat abstract image format

File: pkg/image/image.go (hunk 2 above)
Comment (cclerget) [reply pkg/image/image.go]: yep will change that. Thanks !

**Review by mem — APPROVED** [review]
thanks!

**Comment by ikaneshiro** [comment]
@cclerget seems like you should add a `HasRootFs()` check to this function as well
https://github.com/sylabs/singularity/blob/master/cmd/internal/cli/actions_linux.go#L40

**Pull request merged**